src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -96,6 +101,8 @@ export class OrganizationIntegrations extends AsyncComponent<
       ...super.getDefaultState(),
       list: [],
       displayedList: [],
+      selectedCategory: '',
+      categoriesList: [],
```

Comment:
@NisanthanNanthakumar why do we need this in state? Can't we figure this out from the list of integrations we have?

Reply:
@scefali I'm storing the `categories` in state, so that we only have calculate the values once.

Reply:
@NisanthanNanthakumar I don't think there is really much performance benefit in doing this. Recalculating the categories is not a computationally expensive task.

Reply:
@scefali sure, but `categories` is not a dynamic value that needs to be recalculated on every dropdown click. It's pretty constant. I'm not sure what the best practice here.

Reply:
@NisanthanNanthakumar

I prefer minimalist state in general. I’ll leave this to you as an optional suggestion.

Reply:
👍 on doing JIT computation of simple things vs storing in state. Less state to keep in mind.